stubs/transformers-stubs/models/auto/tokenization_auto.pyi

```diff
@@ -1,21 +1,39 @@
 import os
-import logging
-from typing import Dict, Optional, OrderedDict, Sequence, Tuple, Union
-from .configuration_auto import replace_list_option_in_docstrings
-from transformers.tokenization_utils_fast import PreTrainedTokenizerFast
-from transformers.tokenization_utils import PreTrainedTokenizer
+from collections import OrderedDict
+from typing import Any
+
 from transformers.configuration_utils import PretrainedConfig
+from transformers.tokenization_utils import PreTrainedTokenizer
+from transformers.tokenization_utils_fast import PreTrainedTokenizerFast
+
+from .auto_factory import _LazyAutoMapping
 
-logger: logging.Logger
-TOKENIZER_MAPPING: OrderedDict[PretrainedConfig, Tuple[PreTrainedTokenizer, PreTrainedTokenizerFast]] = ...
-NO_CONFIG_TOKENIZER: Sequence[Union[PreTrainedTokenizer, PreTrainedTokenizerFast]] = ...
-SLOW_TOKENIZER_MAPPING: Dict[PretrainedConfig, Union[PreTrainedTokenizer, PreTrainedTokenizerFast]] = ...
+TOKENIZER_MAPPING_NAMES: OrderedDict[str, tuple[str|None, str|None]]
+TOKENIZER_MAPPING: _LazyAutoMapping
+CONFIG_TO_TYPE: dict[type[PretrainedConfig], str]
 
-def tokenizer_class_from_name(class_name: str) -> Union[PreTrainedTokenizer, PreTrainedTokenizerFast]: ...
+def tokenizer_class_from_name(class_name: str) -> PreTrainedTokenizer| PreTrainedTokenizerFast: ...
 
-def get_tokenizer_config(pretrained_model_name_or_path: Union[str, os.PathLike], cache_dir: Optional[Union[str, os.PathLike]] = ..., force_download: bool = ..., resume_download: bool = ..., proxies: Optional[Dict[str, str]] = ..., use_auth_token: Optional[Union[bool, str]] = ..., revision: Optional[str] = ..., local_files_only: bool = ..., **kwargs) -> Dict: ...
+def get_tokenizer_config(
+    pretrained_model_name_or_path: str | os.PathLike[str], 
+    cache_dir: str | os.PathLike[str] | None = None, 
+    force_download: bool = False,
+    resume_download: bool = False, 
+    proxies: dict[str, str] | None = None, 
+    use_auth_token: bool | str | None = None,
+    revision: str | None = None, 
+    local_files_only: bool = False,
+    subfolder: str = "",
+    **kwargs
+) -> dict[str, Any]: ...
 
 class AutoTokenizer:
     def __init__(self) -> None: ...
     @classmethod
-    def from_pretrained(cls, pretrained_model_name_or_path: Union[str, os.PathLike] = ..., *inputs, **kwargs) -> Union[PreTrainedTokenizer, PreTrainedTokenizerFast]: ...
+    def from_pretrained(
+        cls,
+        pretrained_model_name_or_path: str | os.PathLike[str], 
```

Comment:
Should we be including the default value for `pretrained_model_name_or_path`? Or does it not have one (i.e. `...` in old version was wrong)?

Reply:
It doesn't have one. Source:
https://github.com/huggingface/transformers/blob/70c79940957fb25b54bd1b106935c756b90345eb/src/transformers/models/auto/tokenization_auto.py#L529